com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Camera/HDCamera.cs

```diff
@@ -357,6 +385,134 @@ internal GameObject exposureTarget
+                // 3 elements. Not necesary when running in player mode, since the requests get updated properly (due to swap chain complexities)
+                request.Update();
+#endif
+                if (!request.done && !request.hasError)
+                    break;
+
+                // If this has an error, just skip it
+                if (!request.hasError)
+                {
+                    // Grab the native array from this readback
+                    NativeArray<float> exposureValue = request.GetData<float>();
+                    if (requestState.isDeExposure)
+                        m_GpuDeExposureValue = exposureValue[0];
+                    else
+                        m_GpuExposureValue   = exposureValue[0];
+                }
+                m_ExposureAsyncRequest.Dequeue();
+            }
+        }
+
+        // This function processes the asynchronous read-back requests for the exposure and updates the last known exposure value.
+        internal float GpuExposureValue()
+        {
+            PumpReadbackQueue();
+            return m_GpuExposureValue;
+        }
+
+        // This function processes the asynchronous read-back requests for the exposure and updates the last known exposure value.
+        internal float GpuDeExposureValue()
+        {
+            PumpReadbackQueue();
+            return m_GpuDeExposureValue;
+        }
+
+        internal struct ExposureTextures
+        {
+            public bool useCurrentCamera;
+            public RTHandle current;
+            public RTHandle previous;
+
+            public void clear()
+            {
+                current = null;
+                previous = null;
+            }
+        }
+
+        private bool m_ExposureControlFS = false;
+        internal bool exposureControlFS { get { return m_ExposureControlFS; } }
+        private ExposureTextures m_ExposureTextures = new ExposureTextures(){ useCurrentCamera = true, current = null, previous = null};
+        internal ExposureTextures currentExposureTextures { get { return m_ExposureTextures; } }
+
+        internal void SetupExposureTextures()
+        {
+            if (!m_ExposureTextures.useCurrentCamera)
+                return;
+
+            if (!m_ExposureControlFS)
+            {
+                m_ExposureTextures.current = null;
+                m_ExposureTextures.previous = null;
+                return;
+            }
+
+            var currentTexture = GetCurrentFrameRT((int)HDCameraFrameHistoryType.Exposure);
+            if (currentTexture == null)
+            {
+                RTHandle Allocator(string id, int frameIndex, RTHandleSystem rtHandleSystem)
+                {
+                    // r: multiplier, g: EV100
+                    var rt = rtHandleSystem.Alloc(1, 1, colorFormat: HDRenderPipeline.k_ExposureFormat,
+                        enableRandomWrite: true, name: $"{id} Exposure Texture {frameIndex}"
+                    );
```

Comment:
I guess this code need to be updated with this fix: https://github.com/Unity-Technologies/Graphics/pull/3867 can you check?

Reply:
Also Poking @FrancescoC-unity  just to have a quick check of this PR

Reply:
Re: #3867 the textures are different ones (my PR is on the artist driven exposure curve).

Will make a pass on the rest of PR soon

Reply:
The exposure curve PR should be fine. Will ask QA to test all exposure modes.